### PR TITLE
ENH:made default tick formatter to switch to scientific notation earlier

### LINKED ISSUE #14800: default tick formatter could switch to scientific notation earlier
### Bug report

**Bug summary**

The default tick formatter switches to scientific notation only for very small values, but there is a range of slightly bigger values where scientific notation would be better too.

**Code for reproduction**

```python
import matplotlib.pyplot as plt
plt.rcdefaults()
axs = plt.figure().subplots(2)
axs[0].set(xlim=(1e-7, 21e-7))
axs[1].set(xlim=(1e-8, 21e-8))
plt.show()
```

**Actual outcome**

test

**Expected outcome**

Either the top axes should also use scientific notation (note that this also happens e.g. 1e-6 and 1e-5), or we need something smarter to determine the number of ticks.
IIRC the default ticker places ticks assuming that labels will have a width of no more than ~3 character heights(?), so allowing for 4-5 digits before switching to scientific notation may make sense.

**Matplotlib version**

  * Operating system: linux
  * Matplotlib version: head, at least as far back as 3.0
  * Matplotlib backend (`print(matplotlib.get_backend())`): qt5agg
  * Python version: 37
  * Jupyter version (if applicable):
  * Other libraries:

### PR DESCRIPTION
**Pull request by sameshl**
The default tick formatter switches to scientific notation only for very small values,
but there is a range of slightly bigger values where scientific notation would be better too.

changed the formatter limits from -7,7 to -5,5

closes #14800

## PR Checklist

- [ ] Has Pytest style unit tests
- [x] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by jklymak** [comment]
This prob needs an API change note, and you probably need to edit the default `.matplotlibrc`?

File: doc/api/prev_api_changes/api_changes_3.1.0.rst [hunk 1]
```diff
@@ -1157,3 +1157,8 @@ result in undefined behavior.  It now throws a `ValueError`.
 
 The signature of the (private) ``Axis._update_ticks`` has been changed to not
 take the renderer as argument anymore (that argument is unused).
+
+Reduced default value of `axes.formatter.limits`
```
Comment (jklymak) [inline doc/api/prev_api_changes/api_changes_3.1.0.rst]: Sorry, should have been clear about where this goes.  it gets a new file in `doc/api/next-api-changes/`

Comment (sameshl) [reply doc/api/prev_api_changes/api_changes_3.1.0.rst]: @jklymak I am not able to exactly understand the naming convention of the files in `doc/api/next-api-changes/`. For eg, in `2019-05-08-AL.rst` what exactly does AL mean?

Comment (jklymak) [reply doc/api/prev_api_changes/api_changes_3.1.0.rst]: `AL` are the authors initials.  Which is a bit mysterious, i.e. in that case "Antony Lee".

Comment (sameshl) [reply doc/api/prev_api_changes/api_changes_3.1.0.rst]: And the date is the date of modification? Right?

Comment (jklymak) [reply doc/api/prev_api_changes/api_changes_3.1.0.rst]: Yes

**Comment by timhoffm** [comment]
There are two failing image comparison tests because of the change. How do we want to handle these?

I propose to set the defaults back to -7, 7 for these tests. It's not worth or necessary regenerating the images.

**Comment by sameshl** [comment]
> I propose to set the defaults back to -7, 7 for these tests
@timhoffm Do you mean revert the change in matplotlibrc to -5, 5 ?

**Comment by timhoffm** [comment]
Yes, at the beginning of these two tests, just add
~~~
# use former defaults to match existing baseline image
plt.rcParams['axes.formatter.limits'] = -7, 7
~~~

Note: There's no need to revert that setting at the end of the test because each test starts with the default settings.

**Comment by sameshl** [comment]
I see. I made the required changes. Thanks for the info @timhoffm!

**Comment by timhoffm** [comment]
Sorry, I may not have been clear enough.

I don't think we should apply this to all image comparison tests, but only on a case-by-case basis in the actual test functions where it's necessary to get the the tests pass without regenerating the baseline images (IIRC that have been two tests).

**Comment by sameshl** [comment]
Sorry, I think I misunderstood you. But thank you for helping me out.  :+1: I think I have made the required changes

**Comment by tacaswell** [comment]
Seems reasonable, but has anyone thought about what this does to large numbers?  We might want to have asymmetric thresholds.

**Comment by timhoffm** [comment]
@tacaswell Good point. Maybe -`5, 6`? 5000000 is not quite readable anymore either.

**Comment by sameshl** [comment]
Even, I think -5, 6 would be appropriate. I can make that change

**Comment by jklymak** [comment]
... piling on: It'd be nice for a test w the new limits.  It would also clarify the new behaviour and why this limit is better, because otherwise I'm not sure I follow.

**Comment by timhoffm** [comment]
ping @jklymak do you need additional tests?

File: doc/api/next_api_changes/2019-07-18-SL.rst [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Reduced default value of `axes.formatter.limits`
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Changed the default value of `axes.formatter.limits` from -7, 7 to -5, 6 for better readability.
```
Comment (tacaswell) [inline doc/api/next_api_changes/2019-07-18-SL.rst]: @sameshl If you get a chance could you add a `.. plot` example to the API change note?  It greatly helps people who are skimming the release note to understand the change!

**Comment by sameshl** [comment]
@tacaswell I will update api note with the `.. plot` example by tomorrow

**Comment by tacaswell** [comment]
I will open a PR with the example, merged to keep things moving.

Thanks @sameshl !

**Comment by sameshl** [comment]
Thanks for taking up the example @tacaswell !

**Pull request merged**